### ISSUE TITLE
Sending of `Blob` as `data` with `binary: true`

### ISSUE THREAD

**Issue by AlttiRi** [violentmonkey/violentmonkey#1505]
With
```js
binary: true,
body: new Blob(["xxx"]),
```

VM sends `[Object Blob]`:

![image](https://user-images.githubusercontent.com/16310547/169811145-d53c42a5-8d0f-4e30-a13c-72767be15d18.png)

without `binary: true` it sends `data:application/octet-stream;base64,eHh4` _(is it correct?)_

![image](https://user-images.githubusercontent.com/16310547/169813339-3283111c-61a7-4be8-a9fe-dd5ee5aae46d.png)

---

While the expected result is `xxx`:

![image](https://user-images.githubusercontent.com/16310547/169811251-b5e2190d-b85f-4d40-97ab-ac487c993e18.png)

TM _(with and **without** `binary: true`)_, default `XHR` and `fetch` work such way.

---

https://example.com/gm_send_blob_data

```js
// ==UserScript==
// @name         [GM_XHR] binary Blob data (2022.05.23)
// @description  [GM_xmlhttpRequest] binary Blob data (2022.05.23)
// @version      0.0.1
// @namespace    gh.alttiri
// @match        https://example.com/gm_send_blob_data
// @grant        GM_xmlhttpRequest
// @connect      example.com
// ==/UserScript==

GM_xmlhttpRequest({
    binary: true,
    url: "https://example.com",
    method: "post",
    data: new Blob(["xxx"]),
    responseType: "blob",
    onload: response => console.log("[onload]", response.response),
    onreadystatechange: response => console.log("[onload]", response),
    onprogress: ({loaded, total, lengthComputable}) => {console.log({loaded, total, lengthComputable});}
});
```

---

VM Beta from Store, Chrome, Win 10.
(Did not test Firefox)

**Comment by tophf** [comment]
`binary: true` assumes the input to be a string by definition so supplying any other value makes no sense. Why do you do it?

**Comment by tophf** [comment]
It was added only for compatibility with other script engines that couldn't or still can't real binary data.

**Comment by tophf** [comment]
Fixed in 16fb578c.
Test build: [zip](https://github.com/violentmonkey/violentmonkey/suites/6619792160/artifacts/249305715)

In VM you can send binary `data` by default, without specifying `binary` option.
